src/webapi/blob.rs

```diff
@@ -31,6 +34,46 @@ pub trait IBlob: ReferenceType {
             Some( mime )
         }
     }
+
+    /// Create a new `Blob` object containing the data in the specified range of bytes of the
+    /// source `Blob`.
+    /// 
+    /// See also [slice_with_content_type](IBlob::slice_with_content_type).
+    /// 
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/Blob/slice)
+    // https://w3c.github.io/FileAPI/#ref-for-dfn-slice
+    fn slice< T >( &self, range: T ) -> Blob
+        where T: RangeBounds<u64>
+    {
+        self._slice(range, None)
+    }
+
+    /// [slice](IBlob::slice) `Blob` with the provided `content_type`.
+    fn slice_with_content_type< T >( &self, range: T, content_type: &str ) -> Blob
+        where T: RangeBounds<u64>
+    {
+        self._slice(range, Some(content_type))
+    }
+
+    #[doc(hidden)]
+    fn _slice< T >( &self, range: T, content_type: Option<&str> ) -> Blob
+        where T: RangeBounds<u64>
+    {
+        let start: Option<Number> = match range.start_bound() {
+            Bound::Included(&n) => Some(n),
+            Bound::Excluded(&n) => Some(n + 1),
+            _ => None
+        }.try_into().unwrap();
+        let end: Option<Number> = match range.end_bound() {
+            Bound::Included(&n) => Some(n + 1),
+            Bound::Excluded(&n) => Some(n),
+            _ => None
```

Comment:
Please explicitly match `Bound::Unbounded`. You should be able to use something like `.map( |value| value.try_into().unwrap() )` on this `Option` so that the extra `TryFrom` impl you've added will be unnecessary.

Reply:
Need I also replace `None` with something like `self.len()` here? I was assuming it be converted to `undefined` so that the default parameter is used. Was I right?